tests/integration/search_test.t

```diff
@@ -0,0 +1,167 @@
+		expiration_date => "test",
+		ingredients_text => "apple, milk, eggs, palm oil",
+		origin => "france",
+		serving_size => "10g",
+		packaging_text => "no",
+		action => "process",
+		type => "add",
+		".submit" => "submit"
+	},
+	{
+		code => '200000000037',
+		lang => "en",
+		product_name => "vegan & palm oil free",
+		generic_name => "Tester",
+		quantity => "100 ml",
+		link => "https://github.com/openfoodfacts/openfoodfacts-server",
+		expiration_date => "test",
+		ingredients_text => "fruit, rice",
+		origin => "france",
+		serving_size => "10g",
+		packaging_text => "no",
+		action => "process",
+		type => "add",
+		".submit" => "submit"
+
+	},
+	{
+		code => '200000000038',
+		lang => "en",
+		product_name => "palm oil free & non vegan",
+		generic_name => "Tester",
+		quantity => "100 ml",
+		link => "https://github.com/openfoodfacts/openfoodfacts-server",
+		expiration_date => "test",
+		ingredients_text => "apple, milk, eggs",
+		origin => "france",
+		serving_size => "10g",
+		packaging_text => "no",
+		action => "process",
+		type => "add",
+		".submit" => "submit"
+	},
+	{
+		code => '200000000039',
+		lang => "es",
+		product_name => "Vegan Test Snack",
+		generic_name => "Tester",
+		quantity => "100 ml",
+		link => "https://github.com/openfoodfacts/openfoodfacts-server",
+		expiration_date => "test",
+		ingredients_text => "apple, water, palm oil",
+		origin => "spain",
+		serving_size => "5g",
+		packaging_text => "no",
+		action => "process",
+		categories => "snacks",
+		type => "add",
+		".submit" => "submit"
+	},
+	{
+		code => '200000000045',
+		lang => "es",
+		product_name => "Vegan Test Snack",
+		generic_name => "Tester",
+		quantity => "100 ml",
+		link => "https://github.com/openfoodfacts/openfoodfacts-server",
+		expiration_date => "test",
+		ingredients_text => "apple, water",
+		origin => "France",
+		serving_size => "5g",
+		packaging_text => "no",
+		action => "process",
+		categories => "breakfast cereals",
+		type => "add",
+		".submit" => "submit"
+	}
+);
+
+remove_all_users();
+
+remove_all_products();
+
+wait_dynamic_front();
+
+my $ua = new_client();
+
+my %create_user_args = (%default_user_form, (email => 'bob@gmail.com'));
+create_user($ua, \%create_user_args);
+
+foreach my $ref (@products) {
+	create_product($ua, $ref);
+}
+
+my @tests = (
+	["q1", construct_test_url("/cgi/search.pl?action=process&json=1", "world")],
+	["q2", construct_test_url("/cgi/search.pl?action=process&json=1&ingredients_from_palm_oil=without", "world")],
+	[
+		"q3",
+		construct_test_url(
+			"/cgi/search.pl?action=process&code=200000000034,200000000039&fields=code,product_name&json=1", "world"
+		)
+	],
+	[
+		"q4",
+		construct_test_url(
+			"/cgi/search.pl?action=process&tagtype_0=categories&tag_contains_0=contains&tag_0=breakfast_cereals&tagtype_1=nutrition_grades&tag_contains_1=contains&ingredients_from_palm_oil=without&json=1",
+			"world"
+		)
+	]
+);
+
+## TODO : find out whats up with these quesries below :(
```

Comment:
I think it would be a good idea to print more info in the test logs if some query fails.
e.g. like this: https://github.com/openfoodfacts/openfoodfacts-server/pull/7294